FEATURE: improve blank page syndrome

AndrewPrigorshnev opened the pull request:
**Before:**

<img width="650" alt="Screenshot 2022-01-26 at 13 11 31" src="https://user-images.githubusercontent.com/1274517/151417347-01c8685e-5a30-433a-84ac-b577f803806b.png">

**After:**

<img width="650" alt="Screenshot 2022-01-26 at 13 27 03" src="https://user-images.githubusercontent.com/1274517/151417554-56342da1-3e77-47fe-a1e4-784b181e70cf.png">

pmusaraj commented on config/locales/client.en.yml:
```diff
@@ -4,11 +4,18 @@ en:
       docs:
         help: "browse docs topics"
     docs:
+      or: "or"
```
This doesn't usually translate well, I think you're better off using a single string and just including "categories or tags" in the string regardless of what is enabled in docs.

pmusaraj commented on assets/javascripts/discourse/controllers/docs-index.js.es6:
```diff
@@ -194,6 +196,65 @@ export default Controller.extend({
+
+      body += ` (${this.docsCategoriesAndTags.join(", ")}).`;
+    } else {
+      const setUpPluginMessage = I18n.t("docs.no_docs.setup_the_plugin", {
```
We should probably show this to staff only.

pmusaraj commented on assets/javascripts/discourse/templates/docs-index.hbs:
```diff
@@ -1,6 +1,9 @@
-  {{#if emptyTopics}}
-    <span class="no-topics-found">{{html-safe (i18n "docs.no_topics")}}</span>
+  {{#if noContent}}
+    <EmptyState
```
I believe we need to add a `.discourse-compatibility` file here, just to be safe with previous stable versions.

AndrewPrigorshnev replied:
Fixed it like this:

`
2.8.0.beta9: 05678c451caf2ceb192501da91cf0d24ea44c8e8
`

We could actually do 

`
2.8.0.beta8: 05678c451caf2ceb192501da91cf0d24ea44c8e8
`

since the EmptyState component was introduced one beta version earlier.  But then one version later we did [this improvement](https://github.com/discourse/discourse/commit/f97c8ff0a64a7bd15da87067183a1f9ed8402e4f) to the component to support styling in Teams. Using 2.8.0.beta9 in `.discourse-compatibility` would break styling of empty pages in Teams.

AndrewPrigorshnev commented:
@pmusaraj addressed all your comments, they all make sense, thank you for review! 

> This doesn't usually translate well, I think you're better off using a single string and just including "categories or tags" in the string regardless of what is enabled in docs.

Was on the fence with this, but now I see that this totally makes sense. Initially, I wanted to be even more smarter with this message :).

pmusaraj reviewed (approved):
Looks great, thanks!
